source/agora/utils/PrettyPrinter.d

```diff
@@ -274,10 +274,13 @@ private struct TransactionFmt
             enum InputPerLine = 3;
             enum OutputPerLine = 3;
 
-            formattedWrite(sink, "Type : %s, Inputs (%d): %(%(%s, %),\n%)\n",
-                this.value.type,
-                this.value.inputs.length,
-                this.value.inputs.map!(v => InputFmt(v)).chunks(InputPerLine));
+            if (this.value.inputs.length)
+                formattedWrite(sink, "Type : %s, Inputs (%d): %(%(%s, %),\n%)\n",
+                    this.value.type,
+                    this.value.inputs.length,
+                    this.value.inputs.map!(v => InputFmt(v)).chunks(InputPerLine));
+            else
+                formattedWrite(sink, "Type : %s, Inputs: None\n", this.value.type);
```

Comment:
This special case is so that we don't have trailing whitespace at the end of the line, which can be a PITA.